flink-core/src/main/java/org/apache/flink/configuration/DeploymentOptions.java

```diff
@@ -84,4 +84,13 @@
                     .withDescription(
                             "Whether a Flink Application cluster should shut down automatically after its application finishes"
                                     + " (either successfully or as result of a failure). Has no effect for other deployment modes.");
+
+    public static final ConfigOption<Boolean> ALLOW_CLIENT_JOB_CONFIGURATIONS =
+            ConfigOptions.key("execution.allow-client-job-configurations")
```

Comment:
`execution.immutable-configuration`?

Reply:
I think in Flink there is a general misunderstanding of where the configuration is coming from. I miss which configuration is immutable with `immutable-configuration`.